consensus/cryptarchia-sync/network/src/sync_utils.rs

```diff
@@ -0,0 +1,25 @@
+use std::io::Error as IoError;
+
+use futures::AsyncWriteExt;
+use libp2p::{PeerId, Stream, StreamProtocol};
+use libp2p_stream::{Control, OpenStreamError};
+use nomos_core::wire::packing::{pack_to_writer, unpack_from_reader};
+use serde::{de::DeserializeOwned, Serialize};
+
+pub async fn send_data<T: Serialize + Sync>(stream: &mut Stream, data: &T) -> Result<(), IoError> {
+    pack_to_writer(data, stream).await?;
+    stream.flush().await?;
+    Ok(())
+}
+
+pub async fn receive_data<T: DeserializeOwned>(stream: &mut Stream) -> Result<T, IoError> {
+    unpack_from_reader(stream).await
+}
+
+pub async fn open_stream(
+    peer_id: PeerId,
+    control: &mut Control,
+    protocol: StreamProtocol,
+) -> Result<Stream, OpenStreamError> {
+    control.open_stream(peer_id, protocol).await
+}
```

Comment:
This probably would benefit from `#[inline]` although the compiler should be smart enough.